lib/smart_proxy.rb

```diff
@@ -150,6 +159,12 @@ def launch
       end
 
       (t1 || t2).join
+    rescue Exception => e
```

Comment:
This causes an issue when `daemon` is enabled...

<pre>
E, [2015-07-13T14:23:58.804946 #20218] ERROR -- : Error during startup, terminating. exit SystemExit
D, [2015-07-13T14:23:58.805110 #20218] DEBUG -- : exit:/usr/share/foreman-proxy/lib/daemon.rb:4:in `exit'
/usr/share/foreman-proxy/lib/daemon.rb:4:in `daemon'
/usr/share/foreman-proxy/lib/smart_proxy.rb:151:in `launch'
/usr/share/foreman-proxy/bin/smart-proxy:6
E, [2015-07-13T14:23:58.809238 #20219] ERROR -- : Error during startup, terminating. exit SystemExit
D, [2015-07-13T14:23:58.809417 #20219] DEBUG -- : exit:/usr/share/foreman-proxy/lib/daemon.rb:6:in `exit'
/usr/share/foreman-proxy/lib/daemon.rb:6:in `daemon'
/usr/share/foreman-proxy/lib/smart_proxy.rb:151:in `launch'
/usr/share/foreman-proxy/bin/smart-proxy:6
</pre>

I think what's happening is that the SystemExit inherits from Exception (rather than RuntimeError, which surprised me) and simply gets caught when the main thread legitimately exits.